src/spinnaker_ros_interface_main.cc

```diff
@@ -246,6 +247,14 @@ void CaptureLoop(CameraPtr pCam) {
     // Begin acquiring images
     pCam->BeginAcquisition();
 
+    // Camera image acquisition time is referenced to when the
+    // camera is started. Here we use an offset time to put the 
+    // time stamps in a system clock referenced frame
+    ros::Time const ros_time_offset = ros::Time::now();
```

Comment:
You don't need to separately store the ros time offset and the FLIR camera offset - you can just measure them on startup, and save the difference. See below.

Reply:
Edit: Actually, this is not only inefficient, it is incorrect: You are measuring ros::Time and the FLIR camera times at different times. This will introduce an offset error.